### PR TITLE
Movie page

### PR DESCRIPTION
**Pull request by Hannes-HH**
Hi guys, thank you for your review. My movie page is finish now. 

+ fetching movie data by genre 
+ implement custom pagination 
+ create cards for the movie page 
+ implement functions to add and remove movies by genre

### CODE REVIEW

File: src/components/Pages/Movies/Movies.js [hunk 1]
```diff
@@ -1,60 +1,75 @@
...
+  const [numOfPages, setNumOfPages] = useState();
+  const genreforURL = useGenre(selectedGenres);
 
-//   const fetchMovies = async () => {
-//     const { data } = await axios.get(
-//       `https://api.themoviedb.org/3/discover/movie?api_key=1e37d8deae00c9b519356f5d9412edbb&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=${page}`
-//     );
-//     console.log(data);
-//   };
+  // const fetchMovies = async () => {
```
Comment (namirsab) [inline src/components/Pages/Movies/Movies.js]: Please remove commented code

File: src/components/Pages/Movies/Movies.js [hunk 2]
```diff
@@ -1,60 +1,75 @@
...
-// };
+  return (
+    <div>
+      <span className="pageTitle">Discover Movies</span>
+      <Genres
+        type="movie"
+        selectedGenres={selectedGenres}
+        setSelectedGenres={setSelectedGenres}
+        genres={genres}
+        setGenres={setGenres}
```
Comment (namirsab) [inline src/components/Pages/Movies/Movies.js:38]: It's not a good practice to pass the `setState` functions directly to children.
Insetad, the children should receive a prop `onGenreSelected` that then the parent (this component) would handle and change the state then.
When a `genre` is selected, this component would:
* set the selected genres accordingly
* set the page to be the first one.

File: src/components/Pages/Movies/Movies.js [hunk 3]
```diff
@@ -1,60 +1,75 @@
...
+              id={c.id}
+              poster={c.poster_path}
+              title={c.title || c.name}
+              date={c.first_air_date || c.release_date}
+              media_type="movie"
+              vote_average={c.vote_average}
+            />
+          ))}
+      </div>
+      {numOfPages > 1 && (
```
Comment (namirsab) [inline src/components/Pages/Movies/Movies.js:55]: Very good! conditional rendering well applied here :)

File: src/components/Pages/Movies/Movies.js [hunk 4]
```diff
@@ -1,60 +1,75 @@
...
+        <CustomPagination setPage={setPage} numOfPages={numOfPages} />
+      )}
+    </div>
+  );
+};
 
-// export default Movies;
+export default Movies;
+
+// 1e37d8deae00c9b519356f5d9412edbb
```
Comment (namirsab) [inline src/components/Pages/Movies/Movies.js]: hm, what's that ?

File: src/hooks/useGenre.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+const useGenre = (selectedGenres) => {
```
Comment (namirsab) [inline src/hooks/useGenre.js]: Good intention here to create a custom hook, but:

* A custom hook should use hooks inside it's function body.
* If that's not the case, then this is not a hook, just a utility function that given an array of genres, get's you back a string of genre ids

I'd move this to a utilities folder, and change the name to something like `getGenresIdsAsString` or something like that.

File: src/hooks/useGenre.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+  if (selectedGenres.length < 1) return "";
+
+  const GenreIds = selectedGenres.map((g) => g.id);
```
Comment (namirsab) [inline src/hooks/useGenre.js]: `GenreIds` should be `genreIds`.
Remember, for variables we use `camelCase`

File: src/components/Genres/Genres.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+  const handleRemove = (genre) => {
+    setSelectedGenres(
+      selectedGenres.filter((selected) => selected.id !== genre.id)
+    );
+    setGenres([...genres, genre]);
+    setPage(1);
+  };
+
+  useEffect(() => {
```
Comment (namirsab) [inline src/components/Genres/Genres.js:27]: If you would move the state up to the parent component, this `useEffect` should be done there.

File: src/components/Genres/Genres.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+      `https://api.themoviedb.org/3/genre/${type}/list?api_key=1e37d8deae00c9b519356f5d9412edbb&language=en-US`
+    )
+      .then((res) => res.json())
+      .then((data) => {
+        setGenres(data.genres);
+      });
+  }, [setGenres, type]);
+
+  return (
+    <div style={{ padding: "6px 0" }}>
```
Comment (namirsab) [inline src/components/Genres/Genres.js]: No inline styling please. Add a `className` and use CSS

File: src/components/Genres/Genres.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+      {selectedGenres.map((genre) => (
+        <Chip
+          style={{ margin: 2 }}
```
Comment (namirsab) [inline src/components/Genres/Genres.js]: Same here, even with material-ui you can use classes and CSS:
https://material-ui.com/customization/components/

**Review by namirsab — CHANGES_REQUESTED** [review]
Hey @Hannes-HH! Your app is looking nice!
I've seen a few things that you can improve. Please take a look!

File: src/components/Pages/Movies/Movies.js (hunk 1 above)
Comment (Hannes-HH) [reply src/components/Pages/Movies/Movies.js]: Removed.

File: src/components/Pages/Movies/Movies.js (hunk 4 above)
Comment (Hannes-HH) [reply src/components/Pages/Movies/Movies.js]: Removed.

File: src/components/Pages/Movies/Movies.js (hunk 2 above)
Comment (Hannes-HH) [reply src/components/Pages/Movies/Movies.js:38]: The children received a prop onGenreSelected with an event handler.

File: src/hooks/useGenre.js (hunk 5 above)
Comment (Hannes-HH) [reply src/hooks/useGenre.js]: I removed the folder name to your suggestion.

File: src/hooks/useGenre.js (hunk 6 above)
Comment (Hannes-HH) [reply src/hooks/useGenre.js]: I changed in genreIds.

File: src/components/Genres/Genres.js (hunk 8 above)
Comment (Hannes-HH) [reply src/components/Genres/Genres.js]: The inline styling is removed.

File: src/components/Genres/Genres.js (hunk 9 above)
Comment (Hannes-HH) [reply src/components/Genres/Genres.js]: The inline styling is removed.

File: src/components/Genres/Genres.js (hunk 7 above)
Comment (Hannes-HH) [reply src/components/Genres/Genres.js:27]: Namir, I don't know what do you mean. Please let's talk about.

**Pull request merged**